custom_components/steam_wishlist/strings.json

```diff
@@ -15,5 +15,15 @@
         "title": "Setup"
       }
     }
+  },
+  "options": {
+    "step": {
+      "init": {
+        "title": "Options",
+        "data": {
+          "show_all_wishlist_items": "Display non-sale wishlist items too in Upcoming Media Card (reboot required)"
```

Comment:
This does not actually require a reboot and will be updated instantly, although the sensor won't be updated until the next time the entity is scheduled to be updated. So that note can be removed.
```suggestion
          "show_all_wishlist_items": "Display non-sale wishlist items too in Upcoming Media Card"
```

Reply:
The note can be removed in all translations below too.

Reply:
Another thing to note is that this will likely cause some warnings depending on the size of the user's wish list.

```
Logger: homeassistant.components.recorder.db_schema
Source: components/recorder/db_schema.py:588
integration: Recorder (documentation, issues)
First occurred: 1:51:21 PM (1 occurrences)
Last logged: 1:51:21 PM

State attributes for sensor.steam_wishlist_xxxxxxxxxxxxx exceed maximum size of 16384 bytes. This can cause database performance issues; Attributes will not be stored
```

Reply:
I wish reboot wasn't required after toggling the setting.  Without a reboot, existing configured Upcoming Media Card will still look exactly the same (after toggling the configuration setting).  After a reboot, the card will start displaying all wishlist items (or switch to displaying only a single item).  If you can suggest a way to do it without rebooting (and it isn't too difficult) I would prefer that way instead.  I didn't think it was possible.

Reply:
Just to be clear a reboot isn't required for this change to work.  You just need to wait until the sensor is updated to see the change in the sensor.  If you toggle the option, then manually update the entity via the `homeassistant.update_entity` service you will see the card update.  The sensor only updates once an hour by default.

Reply:
Sorry for the delay writing back.  Unfortunately, it makes no difference after instigating the service call for `homeassistant.update_entity`.  Nor, did it update automatically after an hour (or even 2 hours).  Still same number of items in the sensor's state attributes.  Does it work for you?  In case it does work for you, maybe you might have some ideas why its not for me?   

In case it doesn't work for you either, maybe you wouldn't mind seeing why it's not updating automatically after an hour or via `homeassistant.update_entity`.  I want to make sure others who try this feature don't have the same behavior as me.

![image](https://github.com/boralyl/steam-wishlist/assets/811008/499f4f27-bbf2-4d36-95ce-aaa9e61188c2)

Reply:
I verified both ways, waiting until the sensor updated by itself, and also using the update entity service call.

What does your card config look like?  I had to change the max to a larger number in order to see additional items.  

```yaml
-  type: custom:upcoming-media-card
   entity: sensor.steam_wishlist_72721898833233222
   max: 50
```

Reply:
I was actually wrong.  It didn't automatically update either.  I inadvertely rebooted at some point.  The actual attribute data isn't at all affected by the change no matter how long I wait.  The card settings don't make any difference to the attributes available for the sensor.

![image](https://github.com/boralyl/steam-wishlist/assets/811008/85119b40-5429-4edd-8818-a2b07e01979c)

Reply:
@boralyl is it possible we are talking about 2 different things?  If it is the same thing.  I can try restoring Home Assistant from a previous backup.  Maybe there's something broken on my HA instance.

Reply:
I believe we are talking about the same thing.  Perhaps there is something in one of your outstanding commits that's causing it to not work as expected.  If you push those commits I can test again locally.

Reply:
Thank you so much.  I would love to know what is wrong since it would be a huge benefit to not have to reboot each time.

Reply:
I just pushed those commits to my forked repo.

Reply:
So I confirmed after pulling in your commits the card no longer updates.  So the good news is it's just a bug as it worked prior to the commits.  I have a pretty good idea of what's causing it, but am looking at the best way to fix it.

Reply:
I really appreciate it!  If you could let me know what was causing the issue; and, your respective fix, I would really appreciate.

Reply:
This also means we can now remove passing that boolean down so that it can get to the util function.  I didn't  make those changes in my commit, so feel free to get rid of passing that param down.

Reply:
Yeah, admittedly, I used a really kludgy way to switch.  Your way is so much nicer.  Ill do it as soon as I get time again.  thanks again.

Reply:
I tested it.  It works very well.  I just pushed the changes.  Let me know if I missed anything.